src/raw/mod.rs

```diff
@@ -1469,7 +1530,7 @@ impl<T> Drop for RawIntoIter<T> {
             }
 
             // Free the table
-            if let Some((ptr, layout)) = self.alloc {
+            if let Some((ptr, layout)) = ManuallyDrop::take(&mut self.table).into_alloc() {
```

Comment:
This isn't great, as `ManuallyDrop::take` only landed in Rust 1.42. I'm not quite sure how to work around this, except by making it an `Option`?